### PR TITLE
[CCDM] DCAT-2098: configurable products

### PR DESCRIPTION
**Pull request by mslabko**
## Purpose of this pull request

This pull request (PR) extends OpenAPI specification with examples for Configurable Product creation workflow.

## Affected pages

- https://developer-stage.adobe.com/commerce/services/composable-catalog/data-ingestion/api-reference/

### CODE REVIEW

**Comment by dshevtsov** [comment]
@mslabko since this is a public repository, links to the internal resources are not recommended. Please add summary of the proposed changes according to the template provided in the description.

**Comment by dshevtsov** [comment]
@mslabko I updated the PR description. Please use similar format in future pull requests here.

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 1]
```diff
@@ -302,10 +302,26 @@ paths:
...
       description: |
+        <h3>Simple Products</h3>
         Create products or replace existing products with a specified `sku` and `scope`.
         If a product with the same data exists with the same SKU and scope, the product update request is ignored.
 
-        Use the update operation to modify values for an existing product.
+        Use the [update operation](#operation/ProductsPatch) to modify values for an existing product.
+        
+        <h3>Configurable Products</h3>
+        Configurable Product represents a product with multiple configurable options (e.g., "color", "size"). Unique intersection of option values, like "color green" and "size Large" defines a Product Variant - independent product with its own SKU, price, and inventory.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:312]:
```suggestion
        A configurable product is a product that offers multiple options, such as color and size. Each unique combination of these options, like 'color green' and 'size large,' defines a product variant. This variant is an independent product with its own SKU, price, and inventory. The configurable product acts as a container for these product variants.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 2]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        Configurable Product serves as a container for Product Variants.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:313]:
```suggestion
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 3]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        
+        To create Configurable Product you need the following:
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:315]:
```suggestion
        To create a configurable product, you need the following:
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 4]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        - [Product Attributes](#operation/ProductMetadataPut): Define the product attributes characteristics used to define a Configurable Product, such as "color", "size", etc.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:316]:
```suggestion
        - [Product attributes](#operation/ProductMetadataPut): Define the product attributes used to define a configurable product, such as "color", "size", etc.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 5]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        - [Configurable Product](#operation/ProductsPut): Define Configurable Product. [Options](#operation/ProductsPut!path=options&t=request) is a required field. It defines the choices available for selection to shopper.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:317]:
```suggestion
        - [Configurable product](#operation/ProductsPut): Define the configurable product. [Options](#operation/ProductsPut!path=options&t=request) is a required field. It defines the choices that a shopper can select.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 6]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        - [Product Variant](#operation/ProductsPut): Define Product Variant for given Configurable Product. ["Attributes"](#operation/ProductsPut!path=attributes) with ["variantReferenceId"](#operation/ProductsPut!path=attributes/variantReferenceId&t=request) and ["links"](#operation/ProductsPut!path=links&t=request) with type `VARIANT_OF` must be defined for Product Variant.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:318]:
```suggestion
        - [Product variant](#operation/ProductsPut): Define a product variant for a configurable product. ["Attributes"](#operation/ProductsPut!path=attributes) with ["variantReferenceId"](#operation/ProductsPut!path=attributes/variantReferenceId&t=request) and ["links"](#operation/ProductsPut!path=links&t=request) of type `VARIANT_OF` must be defined for product variants.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 7]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        
+        Each Product Variant links back to the Configurable Product through its `variantReferenceId`, which corresponds to specific `options[].values[].id` in the Configurable Product.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:320]:
```suggestion
        Each product variant links back to the configurable product through its `variantReferenceId`, which corresponds to specific `options[].values[].id` in the configurable product.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 8]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        
+        To unassign Product Variant from Configurable Product you may do one of the following:
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:322]:
```suggestion
        To unassign a product variant from a configurable product, do one of the following:
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 9]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        - Using [Delete Product API](#operation/ProductsDelete) delete Product Variant
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:323]:
```suggestion
        - Use [Delete Product API](#operation/ProductsDelete) to delete the product variant.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 10]
```diff
@@ -302,10 +302,26 @@ paths:
...
+        - Using [Update Product API](#operation/ProductsPatch) set ["variantReferenceId"](#operation/ProductsPut!path=attributes/variantReferenceId&t=request) to `null` and unassign Product Variant from Configurable Product by removing ["links"](#operation/ProductsPut!path=links&t=request) association
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:324]:
```suggestion
        - Use [Update Product API](#operation/ProductsPatch) to set the ["variantReferenceId"](#operation/ProductsPut!path=attributes/variantReferenceId&t=request) to `null` and unassign the product variant from the configurable product by removing the ["links"](#operation/ProductsPut!path=links&t=request) association.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 11]
```diff
@@ -761,9 +777,15 @@ paths:
         - Prices
       summary: Create prices
       description: |
+        <h3>Simple Products</h3>
         Create or replace existing product prices.
 
         A price must be associated with at least the default price book with id `main`.
+        
+        <h3>Configurable Products</h3>
+        Configurable product price is calculated based on the price of the selected Product Variant.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:786]:
```suggestion
        Configurable product price is calculated based on the price of the selected product variant.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 12]
```diff
@@ -761,9 +777,15 @@ paths:
...
+        Any price passed for Configurable Product SKU will be ignored.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:787]:
```suggestion
        Any price passed for a configurable product SKU will be ignored.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 13]
```diff
@@ -1311,7 +1333,7 @@ components:
             $ref: '#/components/schemas/ProductRoutes'
         options:
           type: array
-          description: Options associated with the product.
+          description: Composite products such as Configurable Product must define a list of product options available for selection to Shopper, e.g. "color", "size", etc.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1336]:
```suggestion
          description: Composite products, such as configurable products, must provide a list of product options that a shopper can select (for example, "color", "size", etc.).
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 14]
```diff
@@ -1378,7 +1400,9 @@ components:
             $ref: '#/components/schemas/ProductImage'
         links:
           type: array
-          description: A list of linked SKUs.
+          description: |
+            A list of linked SKUs. For Product Variant, this is required field that establishes a link between a Product Variant and the corresponding Configurable Product.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1404]:
```suggestion
            A list of linked SKUs. For product variants, this is a required field that establishes a link between a product variant and the corresponding configurable product.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 15]
```diff
@@ -1378,7 +1400,9 @@ components:
...
+            `VARIANT_OF` link type must be specified to establish connection to the SKU of the Configurable Product.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1405]:
```suggestion
            `VARIANT_OF` link type must be specified to establish a connection to the configurable product SKU.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 16]
```diff
@@ -1558,9 +1582,14 @@ components:
...
         values:
           type: array
-          description: A list of value(s) associated with a specified attribute type.
+          description: A list of value(s) associated with a specified attribute code.
           items:
             type: string
+        variantReferenceId:
+          type: string
+          description: |
+            The Variant Reference ID establishes a link between a Product Variant and the corresponding [Option Value ID](#operation/ProductsPut!path=options/values/id&t=request) in a Configurable Product.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1591]:
```suggestion
            The variant reference ID establishes a link between a product variant and the corresponding [Option Value ID](#operation/ProductsPut!path=options/values/id&t=request) in a configurable product.
```

**Review by dshevtsov — COMMENTED** [review]
The link format such as `#operation/ProductsPut!path=attributes/variantReferenceId&t=request` doesn't seem to be working in the local rendering. Is this format required by the internal product requirements?
Examples of a working link `#tag/Products/operation/ProductsPut`, `#operation/ProductsPatch`

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 17]
```diff
@@ -1558,9 +1582,14 @@ components:
...
+            Variant reference ID must be defined only for Product that represent Variant of Configurable Product.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1592]:
```suggestion
            Variant reference ID must be defined only for a product that represents a variant of a configurable product.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 18]
```diff
@@ -1602,56 +1631,56 @@ components:
       properties:
         id:
           type: string
-          description: Product Option ID
+          description: |
+            Product Option ID. For `CONFIGURABLE` or `SWATCH` option type, must be equal ["attribute code"](#operation/ProductsPut!path=attributes/code&t=request) used for Configurable Product, e.g. "color"
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1635]:
```suggestion
            Product option ID. For `CONFIGURABLE` or `SWATCH` option types, this ID must be match the ["attribute code"](#operation/ProductsPut!path=attributes/code&t=request) used for the configurable product (for example, "color").
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 19]
```diff
@@ -1602,56 +1631,56 @@ components:
...
         label:
           type: string
           description: Option label
         required:
           type: boolean
-          description: Indicates whether selecting the option is required.
+          description: Indicates whether selecting the option is required by Shopper in order to add product to the Shopping Cart.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1641]:
```suggestion
          description: Indicates whether selecting the option is required by a shopper to add a product to the shopping cart.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 20]
```diff
@@ -1602,56 +1631,56 @@ components:
...
             Option type. Indicates the product type the option can be assigned to.
-            Merchants can define custom types in addition to the predefined values.
             - `CONFIGURABLE`: Configurable product option
-            - `BUNDLE`: Bundle product option
-            - `SWATCH`: Swatch product option
+            - `SWATCH`: Swatch product option. Must be used for color or text swatches attributes
         values:
           type: array
-          description: A list of option values.
+          description: A list of option values. Defines option values available to Shopper, for example "red" color or "Large" size
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1657]:
```suggestion
          description: A list of option values. Defines option values available to shoppers (for example, "red" color or "large" size).
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 21]
```diff
@@ -1602,56 +1631,56 @@ components:
...
       required:
         - id
-        - sku
       properties:
         id:
           type: string
-          description: Option value ID. For "CONFIGURABLE" type, must be equal to the "attribute code".
+          description: |
+            Option value ID.
+            For `CONFIGURABLE` or `SWATCH` must be equal ["variantReferenceId"](#operation/ProductsPut!path=attributes/variantReferenceId&t=request) defined in Product Variant.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1670]:
```suggestion
            For `CONFIGURABLE` or `SWATCH` option types, this ID must match the ["variantReferenceId"](#operation/ProductsPut!path=attributes/variantReferenceId&t=request) defined in the product variant.
```

File: src/openapi/data-ingestion-schema-v1.yaml [hunk 22]
```diff
@@ -1661,10 +1690,12 @@ components:
       properties:
         type:
           type: string
-          description: Product link type. Merchants can define custom types in addition to the predefined values.
+          description: |
+            Product link type. Merchants can define custom types in addition to the predefined values.
+            `VARIANT_OF` link type must be specified to establish connection to the SKU of the Configurable Product.
```
Comment (jeff-matthews) [inline src/openapi/data-ingestion-schema-v1.yaml:1695]:
```suggestion
            `VARIANT_OF` link type must be specified to establish a connection to the configurable product SKU.
```

**Review by jeff-matthews — COMMENTED** [review]
Editorial suggestions

**Comment by mslabko** [comment]
Thanks @jeff-matthews for suggestions, everything is applied!
@dshevtsov thanks for changing description. Any concerns regarding links or we good to merge?

**Review by le0n4ik — APPROVED** [review]

**Review by dshevtsov — APPROVED** [review]

**Review by jeff-matthews — APPROVED** [review]

**Pull request merged**